lib/api/model/events.dart

```diff
@@ -616,9 +616,14 @@ class ChannelDeleteEvent extends ChannelEvent {
   @JsonKey(includeToJson: true)
   String get op => 'delete';
 
-  final List<ZulipStream> streams;
+  final List<ZulipStreamId>? streams; // TODO(server-10): remove
+  final List<int>? streamIds; // TODO(server-10): remove nullability
 
-  ChannelDeleteEvent({required super.id, required this.streams});
+  ChannelDeleteEvent({
+    required super.id,
+    required this.streams,
+    required this.streamIds,
+  }) : assert(streams != null || streamIds != null);
```

Comment:
We try to reserve `assert` for invariants that are our own responsibility, i.e. those that won't be broken except for some broken logic in the client. Here, the invariant exists, but it's one that can be broken by something out of our control, in particular a badly-behaving server.

Also, `assert`s don't run in production, so this won't work as ["crunchy shell" validation](https://github.com/zulip/zulip-mobile/blob/main/docs/architecture/crunchy-shell.md). It makes sense to want such validation, so the "soft center" of the app can rely on this invariant. But let's do it in `ChannelDeleteEvent.fromJson`; for an example to follow, see `DeleteMessageEvent.fromJson`.